src/chatlog/content/text.cpp

```diff
@@ -337,30 +337,36 @@ QString Text::extractSanitizedText(int from, int to) const
         return "";
 
     QString txt;
-    QTextBlock block = doc->firstBlock();
 
-    for (QTextBlock::Iterator itr = block.begin(); itr != block.end(); ++itr) {
-        int pos =
-            itr.fragment()
-                .position(); // fragment position -> position of the first character in the fragment
-
-        if (itr.fragment().charFormat().isImageFormat()) {
-            QTextImageFormat imgFmt = itr.fragment().charFormat().toImageFormat();
-            QString key = imgFmt.name(); // img key (eg. key::D for :D)
-            QString rune = key.mid(4);
-
-            if (pos >= from && pos < to) {
-                txt += rune;
-                ++pos;
-            }
-        } else {
-            for (QChar c : itr.fragment().text()) {
-                if (pos >= from && pos < to)
-                    txt += c;
-
-                ++pos;
+    for (QTextBlock block = doc->findBlock(from); block.isValid(); block = block.next()) {
+        for (QTextBlock::Iterator itr = block.begin(); itr != block.end(); ++itr) {
+            int pos =
+                itr.fragment()
+                    .position(); // fragment position -> position of the first character in the fragment
+
+            if (itr.fragment().charFormat().isImageFormat()) {
+                QTextImageFormat imgFmt = itr.fragment().charFormat().toImageFormat();
+                QString key = imgFmt.name(); // img key (eg. key::D for :D)
+                QString rune = key.mid(4);
+
+                if (pos >= from && pos < to) {
+                    txt += rune;
+                    ++pos;
+                }
+            } else {
+                for (QChar c : itr.fragment().text()) {
+                    if (pos >= from && pos < to)
+                        txt += c;
+
+                    ++pos;
+                }
             }
         }
+
+        if (block == doc->findBlock(to))
```

Comment:
Could you give me some more details please, because I do not fully understand.

First I would not move `block.isValid()` because it is the needed exit condition for the outer `for`. The last block has an invalid `.next()` member. The `block.isValid()` statement could be replaced by `block != doc->findBlock(to).next()` but this is a very ugly solution (although we get rid of the breaking  `if`) because we end up with an additional linebreak at the and an might run into problems if something goes wrong.
Second the breaking `if`has to be after the inner `for` because the it should parse the last needed block (which is `doc->findBlock(to)`) first and then exit the outer `for`. If I move the breaking `if`inside the inner `for`this could generate too many linebreaks at the end.